src/sagemaker/local/image.py

```diff
@@ -0,0 +1,508 @@
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import errno
+import json
+import logging
+import os
+import platform
+import shlex
+import shutil
+import subprocess
+import sys
+import tempfile
+from subprocess import Popen
+from six.moves.urllib.parse import urlparse
+from time import sleep
+
+import yaml
+
+CONTAINER_PREFIX = "algo"
+DOCKER_COMPOSE_FILENAME = 'docker-compose.yaml'
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+
+class _SageMakerContainer(object):
+    """Handle the lifecycle and configuration of a local docker container execution.
+
+    This class is responsible for creating the directories and configuration files that
+    the docker containers will use for either training or serving.
+    """
+
+    def __init__(self, instance_type, instance_count, image, sagemaker_session=None):
+        """Initialize a SageMakerContainer instance
+
+        It uses a :class:`sagemaker.session.Session` for general interaction with user configuration
+        such as getting the default sagemaker S3 bucket. However this class does not call any of the
+        SageMaker APIs.
+
+        Args:
+            instance_type (str): The instance type to use. Either 'local' or 'local_gpu'
+            instance_count (int): The number of instances to create.
+            image (str): docker image to use.
+            sagemaker_session (sagemaker.session.Session): a sagemaker session to use when interacting
+                with SageMaker.
+        """
+        from sagemaker.local.local_session import LocalSession
+        self.sagemaker_session = sagemaker_session or LocalSession()
+        self.instance_type = instance_type
+        self.instance_count = instance_count
+        self.image = image
+        self.hosts = ['{}-{}'.format(CONTAINER_PREFIX, i) for i in range(1, self.instance_count + 1)]
+        self.container_root = None
+        self.container = None
+        # set the local config. This is optional and will use reasonable defaults
+        # if not present.
+        self.local_config = None
+        if self.sagemaker_session.config and 'local' in self.sagemaker_session.config:
+            self.local_config = self.sagemaker_session.config['local']
+
+    def train(self, input_data_config, hyperparameters):
+        """Run a training job locally using docker-compose.
+        Args:
+            input_data_config (dict): The Input Data Configuration, this contains data such as the
+                channels to be used for training.
+            hyperparameters (dict): The HyperParameters for the training job.
+
+        Returns (str): Location of the trained model.
+        """
+        self.container_root = self._create_tmp_folder()
+        os.mkdir(os.path.join(self.container_root, 'output'))
+
+        data_dir = self._create_tmp_folder()
```

Comment:
data_dir can be a normal tmp directory since is not going to be a docker volume.